stl/inc/format

```diff
@@ -1353,22 +1353,71 @@ _NODISCARD constexpr bool _In_bounds(const _Ty _Value) {
     }
 }
 
+_NODISCARD inline int _Count_separators(size_t _Digits, const string_view _Groups) {
+    int _Separators = 0;
+    if (!_Groups.empty()) {
+        // Calculate the amount of separators that are going to be inserted based on the groupings of the locale.
+        auto _Group_it = _Groups.begin();
+        while (_Digits > static_cast<size_t>(*_Group_it)) {
+            _Digits -= static_cast<size_t>(*_Group_it);
+            ++_Separators;
+            if (_Group_it + 1 != _Groups.end()) {
+                ++_Group_it;
+            }
+        }
+    }
+    return _Separators;
+}
+
+template <class _CharT, class _OutputIt>
+_OutputIt _Write_separated_integer(const char* _Start, const char* const _End, const string_view _Groups,
+    const _CharT _Separator, int _Separators, _OutputIt _Out) {
+    auto _Group_it = _Groups.begin();
+    auto _Repeats  = 0;
+    auto _Grouped  = 0;
+
+    for (int _Section = 0; _Section < _Separators; ++_Section) {
+        _Grouped += *_Group_it;
+        if (_Group_it + 1 != _Groups.end()) {
+            ++_Group_it;
+        } else {
+            ++_Repeats;
+        }
+    }
+    _Out   = _STD copy(_Start, _End - _Grouped, _Out);
+    _Start = _End - _Grouped;
+
+    for (; _Separators > 0; --_Separators) {
+        if (_Repeats > 0) {
+            --_Repeats;
+        } else {
+            --_Group_it;
+        }
+
+        *_Out++ = _Separator;
+        _Out    = _STD copy(_Start, _Start + *_Group_it, _Out);
+        _Start += *_Group_it;
+    }
+    _STL_ASSERT(_Start == _End, "separator writing failed");

```

Comment:
Why do we feel the need to check this function's correctness, but only when compiling in Debug mode? Did you perhaps mean to always check with `_STL_VERIFY`, or check when testing the STL with `_STL_INTERNAL_CHECK`?

Reply:
Probably testing makes the most sense.